### DISCUSSION TITLE
gnome-shell process using 100% CPU

### DISCUSSION THREAD

**Discussion by ldesousa** [regolith-linux/regolith-desktop#1011] [Q&A]
This is happening for about 10/12 days, the `gnome-shell` process is always up there on 100% CPU. There is nothing obvious in the `strace` report (to me). Can this behaviour be in any way related to Regolith? Is there any programme/functionality dependent on this process?

Thank you. 

```bash
$ sudo strace -c -p 1951634
strace: Process 1951634 attached
^Cstrace: Process 1951634 detached
% time     seconds  usecs/call     calls    errors syscall
------ ----------- ----------- --------- --------- ----------------
 21,63    1,698058           2    691926           write
 19,66    1,543766           4    345955           sendmsg
 19,62    1,540796           4    353445           poll
 16,72    1,312641           3    345955           recvmsg
 11,20    0,879005           2    345968           read
 11,03    0,865908           2    345955           epoll_wait
  0,07    0,005251           8       598           mprotect
  0,04    0,003528          15       226        13 futex
  0,02    0,001764          13       133        14 ioctl
  0,00    0,000236           3        60        46 newfstatat
  0,00    0,000140          14        10           mmap
  0,00    0,000065           2        30           getpid
  0,00    0,000062           3        16           getrusage
  0,00    0,000053           5        10           munmap
  0,00    0,000030           7         4           close
  0,00    0,000030           7         4           timerfd_create
  0,00    0,000028           2        12           kcmp
  0,00    0,000019          19         1           clone3
  0,00    0,000018          18         1           madvise
  0,00    0,000015           3         4           timerfd_settime
  0,00    0,000006           3         2           rt_sigprocmask
------ ----------- ----------- --------- --------- ----------------
100,00    7,851419           3   2430315        73 total
```

**Comment by kgilmer** [comment]
Hi @ldesousa , can you provide some basic info about the environment? (distro-version, regolith-version)

> Can this behaviour be in any way related to Regolith? 

Certainly possible.  As a "desktop environment" Regolith is designed as a peer to gnome-shell.  We do not integrate with it or call functions or depend upon it, however Regolith is a hybrid system that uses some GTK3 APIs.  These calls do sometimes unintentionally cause interactions with gnome-shell.

> Is there any programme/functionality dependent on this process?

Not that I'm aware of.  The app launcher (or 'desktop executor' as I like to call it) uses some GTK/GDK APIs for lauching apps.  We use `gsettings` and `dconf`.  But there is no source, link, or runtime dependency from Regolith to gnome-shell that I'm aware of.

The `strace` seems like maybe it's logging errors somewhere..did you check `journalctl` to see if that may be where it's going?

**Reply by ldesousa** [reply]
Hi @kgilmer, thank you replying. System info is below. I went through the journal logs but there is nothing suspicious, no errors or any references to gnome.

```
$ uname -a
Linux luis-notebook 6.5.0-28-generic #29~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Apr  4 14:39:20 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

$ apt policy regolith-desktop
regolith-desktop:
  Installed: 4.3.0-1regolith-jammy
  Candidate: 4.3.0-1regolith-jammy
  Version table:
 *** 4.3.0-1regolith-jammy 500
        500 https://regolith-desktop.org/release-ubuntu-jammy-amd64 jammy/main amd64 Packages
        100 /var/lib/dpkg/status 
```
